content/ko/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,172 @@
+content_type: concept
+weight: 50
+---
+
+<!-- overview -->
+{{<glossary_definition term_id="garbage-collection" length="short">}}
+다음과 같은 리소스를 정리한다:
+
+  * [실패한 파드](/ko/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection)
+  * [종료된 잡](/ko/docs/concepts/workloads/controllers/ttlafterfinished/)
+  * [소유자 참조가 없는 오브젝트](#owners-dependents)
+  * [사용되지 않는 컨테이너와 컨테이너 이미지](#containers-images)
+  * [반환 정책이 삭제인 스토리지클래스에 의해 동적으로 생성된 퍼시스턴트볼륨](/ko/docs/concepts/storage/persistent-volumes/#delete)
+  * [Stale 또는 만료된 CertificateSigningRequests (CSRs)](/docs/reference/access-authn-authz/certificate-signing-requests/#request-signing-process) <!-- en 글에서부터 링크가 깨져있어 /docs를 따로 추가-->
+  * {{<glossary_tooltip text="노드" term_id="node">}} 는 다음과 같은 상황에서 삭제된다:
+    * 클러스터가 [클라우드 컨트롤러 매니저](/ko/docs/concepts/architecture/cloud-controller/)를 사용하는 클라우드
+    * 클러스터가 클라우드 컨트롤러 매니저와 유사한 애드온을 사용하는 온프레미스
+  * [노드 리스(Lease) 오브젝트](/ko/docs/concepts/architecture/nodes/#heartbeats)
+
+## 소유자(Owners)와 종속(dependents) {#owners-dependents}
+
+쿠버네티스의 많은 오브젝트는 [*owner references*](/docs/concepts/overview/working-with-objects/owners-dependents/)를 통해 서로 연결되어 있다.
+
+소유자 참조(Owner references)는 컨트롤 플레인에게 어떤 오브젝트가 서로 종속적인지를 알려준다.
+쿠버네티스는 소유자 참조를 사용하여 컨트롤 플레인과 다른 API 클라이언트에게 오브젝트를 삭제하기 전 관련 리소스를 정리하는 기회를 제공한다. 대부분의 경우, 쿠버네티스는 소유자 참조를 자동으로 관리한다.
+
+소유권(Ownership)은 일부 리소스가 사용하는 [레이블과 셀렉터](/ko/docs/concepts/overview/working-with-objects/labels/)
+메커니즘과는 다르다. 예를 들어, 
+`EndpointSlice` 오브젝트를 생성하는 {{<glossary_tooltip text="서비스" term_id="service">}}를 
+생각해보자. 서비스는 *레이블*을 사용해 컨트롤 플레인이
+어떤 `EndpointSlice` 오브젝트가 해당 서비스에 의해 사용되는지 판단하는 데 도움을 준다. 레이블과 더불어,
+서비스를 대신해 관리되는 각 `EndpointSlice` 오브젝트는
+소유자 참조를 가진다. 소유자 참조는 쿠버네티스의 다른 부분이 제어하지 않는
+오브젝트를 방해하는 것을 방지하는 데 도움을 준다.
+
+
+{{< note >}}
+교차 네임스페이스(cross-namespace)의 소유자 참조는 디자인상 허용되지 않는다.
+네임스페이스 종속 오브젝트는 클러스터 범위 또는 네임스페이스 소유자를 지정할 수 있다.
+네임스페이스 소유자는 **반드시** 종속 오브젝트와 동일한 네임스페이스에 존재해야 한다.
+그렇지 않다면, 소유자 참조는 없는 것으로 간주되어, 종속 오브젝트는
+모든 소유자가 없는 것으로 확인되면 삭제될 수 있다.
+
+클러스터 범위의 종속 오브젝트는 클러스터 범위의 소유자만 지정할 수 있다.
+v1.20 이상에서, 클러스터 범위의 종속 오브젝트가 네임스페이스 종류를 소유자로 지정하면,
+확인할 수 없는 소유자 참조가 있는 것으로 간주되어 가비지 수집이 될 수 없다.
+
+v1.20 이상에서, 가비지 수집기가 잘못된 교차 네임스페이스 `ownerReference`
+또는 네임스페이스 종류를 참조하는 `ownerReference`가 있는 클러스터 범위의 종속 항목을 감지하면,
+`OwnerRefInvalidNamespace`가 원인인 경고 이벤트와 유효하지 않은 종속 항목의 `involvedObject`가 보고된다.
+`kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`
+를 실행하여 이러한 종류의 이벤트를 확인할 수 있다.
+{{< /note >}}
+
+## 캐스케이딩(Cascading) 삭제 {#cascading-deletion}
+
+쿠버네티스는 오브젝트를 삭제할 때 더 이상 소유자 참조가 없는지,
+예를 들어 레플리카셋을 삭제할 때, 남겨진 파드가 없는지 확인하고 삭제한다.
+오브젝트를 삭제할 때 쿠버네티스가 오브젝트의 종속 오브젝트들을 자동으로 삭제할 지 여부를 제어할 수 있다.
+이 과정을 `캐스케이딩 삭제`라고 한다.
+캐스케이딩 삭제에는 다음과 같은 두 가지 종류가 있다.
+
+  * 포어그라운드 캐스케이딩 삭제(Foreground cascading deletion)
+  * 백그라운드 캐스케이딩 삭제(Background cascading deletion)
+
+또한 쿠버네티스의 {{<glossary_tooltip text="finalizers" term_id="finalizer">}}를 사용하여 가비지 수집이 소유자 참조가 있는 자원을 언제 어떻게 삭제할 것인지 제어할 수 있다.
+
+### 포어그라운드 캐스케이딩 삭제 {#foreground-deletion}
```

Comment:
AFAIK most IT materials (books, blogs, news articles, ...) use "포그라운드" rather than "포어그라운드". 포그라운드 & 백그라운드 are typical translation terms. Would you change all the words on this PR from "포어그라운드" to "포그라운드"?

Reply:
Sure!